grpc-api/src/main/java/io/deephaven/grpc_api/arrow/FlightServiceGrpcImpl.java

```diff
@@ -0,0 +1,541 @@
+
+    private static final Logger log = LoggerFactory.getLogger(FlightServiceGrpcImpl.class);
+
+    private final SessionService sessionService;
+    private final TicketRouter ticketRouter;
+
+    @Inject()
+    public FlightServiceGrpcImpl(final SessionService sessionService,
+                                 final TicketRouter ticketRouter) {
+        this.sessionService = sessionService;
+        this.ticketRouter = ticketRouter;
+    }
+
+    @Override
+    public void listFlights(final Flight.Criteria request, final StreamObserver<Flight.FlightInfo> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            ticketRouter.visitFlightInfo(responseObserver::onNext);
+            responseObserver.onCompleted();
+        });
+    }
+
+    @Override
+    public void getFlightInfo(final Flight.FlightDescriptor request, final StreamObserver<Flight.FlightInfo> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            responseObserver.onNext(ticketRouter.flightInfoFor(request));
+            responseObserver.onCompleted();
+        });
+    }
+
+    @Override
+    public void getSchema(final Flight.FlightDescriptor request, final StreamObserver<Flight.SchemaResult> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            final ByteString schema = ticketRouter.flightInfoFor(request).getSchema();
+            responseObserver.onNext(Flight.SchemaResult.newBuilder()
+                    .setSchema(schema)
+                    .build());
+            responseObserver.onCompleted();
+        });
+    }
+
+    public void doGetCustom(final Flight.Ticket request, final StreamObserver<InputStream> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            final SessionState session = sessionService.getCurrentSession();
+            final SessionState.ExportObject<BaseTable> export = ticketRouter.resolve(session, request);
+            session.nonExport()
+                    .require(export)
+                    .submit(() -> {
+                        final BaseTable table = export.get();
+
+                        // Send Schema wrapped in Message
+                        final String[] columnNames = table.getDefinition().getColumnNamesArray();
+                        final ColumnSource<?>[] columnSources = table.getColumnSources().toArray(ColumnSource.ZERO_LENGTH_COLUMN_SOURCE_ARRAY);
+                        final FlatBufferBuilder builder = new FlatBufferBuilder();
+                        final int schemaOffset = BarrageSchemaUtil.makeSchemaPayload(builder, columnNames, columnSources, table.getAttributes());
+                        builder.finish(BarrageStreamGenerator.wrapInMessage(builder, schemaOffset, BarrageStreamGenerator.SCHEMA_TYPE_ID));
+                        final ByteBuffer serializedMessage = builder.dataBuffer();
+
+                        final byte[] msgBytes = BarrageData.newBuilder()
```

Comment:
I think we should create a follow-up backlog ticket for hand-building the protobufs.